Content.Shared/MartialArts/Systems/GrabThrownSystem.cs

```diff
@@ -0,0 +1,93 @@
+using Content.Shared.Damage.Systems;
+using Content.Shared.Damage;
+using Content.Shared.Effects;
+using Content.Shared.MartialArts.Components;
+using Content.Shared.Throwing;
+using Robust.Shared.Network;
+using Robust.Shared.Physics.Events;
+using Robust.Shared.Player;
+using System.Numerics;
+
+namespace Content.Shared.MartialArts.Systems;
+
+public sealed class GrabThrownSystem : EntitySystem
+{
+    [Dependency] private readonly DamageableSystem _damageable = default!;
+    [Dependency] private readonly SharedColorFlashEffectSystem _color = default!;
+    [Dependency] private readonly StaminaSystem _stamina = default!;
+    [Dependency] private readonly ThrowingSystem _throwing = default!;
+    [Dependency] private readonly INetManager _netMan = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<GrabThrownComponent, StartCollideEvent>(HandleCollide);
+        SubscribeLocalEvent<GrabThrownComponent, StopThrowEvent>(OnStopThrow);
+    }
+
+    private void HandleCollide(EntityUid uid, GrabThrownComponent component, ref StartCollideEvent args)
+    {
+        if (_netMan.IsClient)   // To avoid effect spam
+            return;
+
```

Comment:
Predict it properly please, ask on discord if you need help.

If anything you might need to run it on client so it looks correct from their POV and have it tell the server.